### PR TITLE
Set `ZIG_COMMAND` env var for downstream

### PR DESCRIPTION
**Pull request by messense**

### CODE REVIEW

**Comment by messense** [comment]
@ravenexp

File: src/zig.rs [hunk 1]
```diff
@@ -188,7 +188,10 @@ impl Zig {
             }
         }
 
-        let mut child = Self::command()?
+        let mut zig = Self::command()?;
+        env::set_var("ZIG_COMMAND", format!("{:?}", zig));
```
Comment (messense) [inline src/zig.rs:192]: It's `"zig"` or `"python3" "-m" "ziglang"`. I think we should get rid of the double quotes.

**Comment by messense** [comment]
On second thought I think doing this in maturin makes more sense since for Windows MSVC we need to use `cargo-xwin` to drive the build, but we can call [`Zig::find_zig`](https://docs.rs/cargo-zigbuild/latest/cargo_zigbuild/zig/enum.Zig.html#method.find_zig) to find zig and set env var for python3-dll-a along the way.

**Comment by ravenexp** [comment]
I think it is useful to set `ZIG_COMMAND` regardless, because there is also the `x86_64-pc-windows-gnu` target which is also supported by `zig cc`.

If someone wants to compile an extension like:

```
maturin build --target x86_64-pc-windows-gnu --zig
```

The build may fail if MinGW binutils `dlltool` is used by `python3-dll-a` instead of `llvm-dlltool` that Zig uses.
It's better to always use `zig dlltool` when `zig cc` is used as the linker.

**Comment by messense** [comment]
@ravenexp Moved the code to https://github.com/PyO3/maturin/pull/919

**Pull request closed without merging**